### PR TITLE
[inductor] Add typing to ir.py 2

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140915

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/dependencies.py [hunk 1]
```diff
@@ -696,9 +696,11 @@ def extract_input_node_reduction_ranges(
 
             if isinstance(op, ComputedBuffer) and len(op.get_reduction_size()) > 0:
                 if reduction_size is None:
-                    reduction_size = op.get_reduction_size()
-                    size = op.get_size()
-                elif reduction_size != op.get_reduction_size() or size != op.get_size():
+                    reduction_size = [*op.get_reduction_size()]
+                    size = [*op.get_size()]
+                elif reduction_size != [*op.get_reduction_size()] or size != [
```
Comment (aorenste) [inline torch/_inductor/dependencies.py:701]: Why convert the returned sequence into a list? Optional[List[T]] should be comparable to Sequence[T]

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -2204,22 +2213,28 @@ def __post_init__(self) -> None:
...
+        idx = self.reindex(vars, reduction_vars)
         values = [inner_fn(idx) for inner_fn in self.inner_fns]
         result = ops.sort(self.dtypes, values, self.stable, self.descending)
         return ops.store(output_name, indexer(idx), result[self.output_index])
 
     def get_reduction_type(self) -> Optional[str]:
         return "sort"
 
-    def get_reduction_size(self) -> Sequence[_IntLike]:
+    def get_reduction_size(self) -> Sequence[sympy.Expr]:
```
Comment (aorenste) [inline torch/_inductor/ir.py:2231]: nit: `Expr`

**Review by aorenste — APPROVED** [review]

File: torch/_inductor/dependencies.py (hunk 1 above)
Comment (jansel) [reply torch/_inductor/dependencies.py:701]: ``
>>> (1, ) == [1, ]
False
>>> 
```

**Pull request closed without merging**